### PR TITLE
[FIX] mail: allow duplicate records with alias for non-admin

### PR DESCRIPTION
**Pull request by yelizariev**
ALIAS_WRITEABLE_FIELDS is used to write alias fields with sudo

Steps to reproduce:

1. Create a user
1.1. Give rights Services -> Project-> Administartor
1.2. Not given rights Administration-> administartor
2. Project app -> open any project -> Action -> duplicate -> it give validation error.

---

opw-2506566

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by yelizariev** [comment]
The field was added in v14 https://github.com/odoo/odoo/commit/c5e6511e8d35a206e966658325ac8dce6a097a48

**Comment by tde-banana-odoo** [comment]
Hello,

Indeed, seems right! As usual, could be great to write a small test (we have alias tests in test_mail_gateway.py I think).

Thanks !

**Review by nle-odoo — APPROVED** [review]
looks good

as a side note: the field is removed by mail.alias.mixin but the error happen when copy_translations try to copy translations.

robodoo delegate+

**Comment by yelizariev** [comment]
@tde-banana-odoo done!

Error without the update:

<details>

```
2021-05-04 13:36:46,731 1 INFO devel odoo.addons.test_mail.tests.test_mail_gateway: ====================================================================== 
2021-05-04 13:36:46,731 1 ERROR devel odoo.addons.test_mail.tests.test_mail_gateway: ERROR: TestMailAlias.test_alias_mixin_copy
Traceback (most recent call last):
  File "/opt/odoo/custom/src/odoo/odoo/tools/cache.py", line 85, in lookup
    r = d[key]
  File "/opt/odoo/custom/src/odoo/odoo/tools/func.py", line 71, in wrapper
    return func(self, *args, **kwargs)
  File "/opt/odoo/custom/src/odoo/odoo/tools/lru.py", line 34, in __getitem__
    a = self.d[obj]
KeyError: ('ir.model.access', <function IrModelAccess.check at 0x7f4842446670>, 6, False, 'mail.alias', 'write', True, (None,))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/odoo/custom/src/odoo/odoo/models.py", line 3695, in write
    fields[0].determine_inverse(real_recs)
  File "/opt/odoo/custom/src/odoo/odoo/fields.py", line 1187, in determine_inverse
    self.inverse(records)
  File "/opt/odoo/custom/src/odoo/odoo/fields.py", line 568, in _inverse_related
    target[field.name] = record_value[record]
  File "/opt/odoo/custom/src/odoo/odoo/models.py", line 5676, in __setitem__
    return self._fields[key].__set__(self, value)
  File "/opt/odoo/custom/src/odoo/odoo/fields.py", line 1127, in __set__
    records.write({self.name: write_value})
  File "/opt/odoo/auto/addons/mail/models/mail_alias.py", line 122, in write
    return super(Alias, self).write(vals)
  File "/opt/odoo/custom/src/odoo/odoo/models.py", line 3570, in write
    self.check_access_rights('write')
  File "/opt/odoo/custom/src/odoo/odoo/models.py", line 3331, in check_access_rights
    return self.env['ir.model.access'].check(self._name, operation, raise_exception)
  File "<decorator-gen-33>", line 2, in check
  File "/opt/odoo/custom/src/odoo/odoo/tools/cache.py", line 90, in lookup
    value = d[key] = self.method(*args, **kwargs)
  File "/opt/odoo/custom/src/odoo/odoo/addons/base/models/ir_model.py", line 1793, in check
    raise AccessError(msg)
odoo.exceptions.AccessError: You are not allowed to modify 'Email Aliases' (mail.alias) records.

This operation is allowed for the following groups:
	- Administration/Settings

Contact your administrator to request access if necessary.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/odoo/auto/addons/test_mail/tests/test_mail_gateway.py", line 176, in test_alias_mixin_copy
    self._test_alias_mixin_copy(user_demo, False)
  File "/opt/odoo/auto/addons/test_mail/tests/test_mail_gateway.py", line 187, in _test_alias_mixin_copy
    record_copy = record.copy()
  File "/opt/odoo/custom/src/odoo/odoo/models.py", line 4662, in copy
    self.with_context(from_copy_translation=True).copy_translations(new, excluded=default or ())
  File "/opt/odoo/custom/src/odoo/odoo/models.py", line 4629, in copy_translations
    new_wo_lang[name] = old_wo_lang[name]
  File "/opt/odoo/custom/src/odoo/odoo/models.py", line 5676, in __setitem__
    return self._fields[key].__set__(self, value)
  File "/opt/odoo/custom/src/odoo/odoo/fields.py", line 1127, in __set__
    records.write({self.name: write_value})
  File "/opt/odoo/auto/addons/mail/models/mail_thread.py", line 322, in write
    result = super(MailThread, self).write(values)
  File "/opt/odoo/auto/addons/mail/models/mail_alias_mixin.py", line 54, in write
    super(AliasMixin, self).write(record_vals)
  File "/opt/odoo/custom/src/odoo/odoo/models.py", line 3699, in write
    raise AccessError(
odoo.exceptions.AccessError: You are not allowed to modify 'Email Aliases' (mail.alias) records.

This operation is allowed for the following groups:
	- Administration/Settings

Contact your administrator to request access if necessary.

Implicitly accessed through 'Project-like model with alias' (mail.test.container).

```

</details>

File: addons/test_mail/tests/test_mail_gateway.py [hunk 1]
```diff
@@ -170,6 +170,23 @@ def test_alias_name_unique(self):
...
             self.env['ir.config_parameter'].sudo().set_param('mail.bounce.alias', new_mail_alias.alias_name)
 
+    def test_alias_mixin_copy(self):
+        user_demo = self.env.ref('base.user_demo')
+        self.assertFalse(user_demo.has_group('base.group_system'), 'Demo user is not supposed to have Administrator access')
+        self._test_alias_mixin_copy(user_demo, 'alias.test1', False)
+        self._test_alias_mixin_copy(user_demo, 'alias.test2', '<p>What Is Dead May Never Die</p>')
+
+    def _test_alias_mixin_copy(self, user, alias_name, alias_bounced_content):
+        record = self.env['mail.test.container'].with_user(user).with_context(lang='en_US').create({
```
Comment (yelizariev) [inline addons/test_mail/tests/test_mail_gateway.py:180]: TIL:  ``self.env.lang`` is False by default in test environment

**Review by nle-odoo — APPROVED** [review]
robodoo r+

**Comment by tde-banana-odoo** [comment]
Hello,

Coming back to this one, please pay attention when writing tests @yelizariev @nle-odoo  . You should not use existing demo data like demo user, but instead use existing data in tests (notably in mail we have plenty of data). You should also avoid sub-methods that make test hard to read (like calling a sub_method with only one parameter that changes). It is better to have a bit more lines but explicit tests. Finally ordering test methods is important as tests are executed alphabetically, meaning they should be sorted in file to have a matching order between failing tests and tests to update when working on features.

In short, even tests require review. Could you pay more attention next time ?

Many thanks,
